### ISSUE TITLE
Orchestration instances not starting (stuck in Pending state)

### ISSUE THREAD

**Issue by ghills** [Azure/azure-functions-durable-extension#540]
We had 40 orchestrator instances that were created in a 'Pending' state but were not actually started. When I noticed the problem it had been over 2 hours for some of them and I restarted the app from the portal which caused them all to be picked up and executed. During this time other orchestration instances were being started and still running successfully.

When I examined this instance I found that the `RuntimeStatus` in `DurableFunctionsHubInstances` was 'Pending' and `DurableFunctionHubHistory` had no records with the instance ID as the partition key.

Once I restarted the app and gave it a minute all of the pending instances were completed and the `DurableFunctionsHubHistory` had the records I expected.

Runtime: v1

Example case (times UTC):
Instance ID: f3b0243177744e819c006b2de3796702
Instance Created: 2018-12-06T15:26:24.688Z
App Restarted (by me in portal): 2018-12-06T17:55:40.454Z
Orchestrator Started: 2018-12-06T17:55:53.927Z

As I write this it looks like we are accumulating more pending instances that have been sitting around for 10-20 minutes. Example instance IDs:

93e522b2b1e042d8a396a98cbeb70c22
b1a34d79f5b5470e9b41068cd10c3f75
ff9dcaf5a5894fb699695b1aa7000d20

In order to track down the function app here is a timestamped function ID (in US West):
> 2018-12-06T18:00:54.648Z	Function started (Id=5c50bb6f-73c5-4e1c-a81a-3529d3b2bf50)

**Comment by ghills** [comment]
@cgillum I looked at #520 (appears to be just what is reported, not what actually ran) and #342 (function host v2) but they didn't quite seem to fit what I was seeing.

**Comment by cgillum** [comment]
@ghills thanks for the detailed information. The fact that a restart fixed the issue likely means that the host somehow got into a weird state where it wasn't picking up new messages.  I'll take a look at our internal telemetry to see if I can find more clues about what happened.

**Comment by cgillum** [comment]
@ghills It looks like you're running a pretty old version of our extension (v1.5.0). I suggest updating to the [latest version](https://github.com/Azure/azure-functions-durable-extension/releases/) (currently v1.7.0) which has a lot of important reliability improvements. I don't know for sure if you're running into a bug that was already fixed, but I recommend this upgrade to rule out existing problems. Upgrading also gives us improved telemetry for future investigations.

**Comment by ghills** [comment]
@cgillum ok I will start down the path of updating that extension package as it looks like the update will be non-trivial for us.

**Comment by hanvyj** [comment]
I'm having this issue running v1.7.0 with the Node framework: https://github.com/Azure/azure-functions-durable-js/issues/32.

Working fine for other developers, but half the time I run a function it is indefinately Pending.

**Comment by ghills** [comment]
@cgillum I have updated the app to v1.7.0 and we haven't seen any instances of this occurring since. It's hard to know for certain as when I rolled out the update is also when our busy time ended and our usage has dropped way down but hopefully it is no longer an issue.

I don't know if you want to go ahead and close this issue out but I can always just bring it back up if I see it happen again. I have implemented a "watchdog" type queue to look out for dangling instances like this so I should be able to catch them if they occur in the future.
